pygsti/modelmembers/operations/opfactory.py

```diff
@@ -502,7 +502,7 @@ def create_op(self, args=None, sslbls=None):
             raise ValueError("Not allowed to embed onto sslbls=" + str(sslbls))
 
         if self.embeds_factory:
-            op = self.embedded_factory_or_op.create_op(args, None)  # Note: will have its gpindices set already
+            op = self.embedded_factory_or_op.create_op(args, sslbls)  # Note: will have its gpindices set already
```

Comment:
I think this change may be fine, but here's why I think I passed `None` as the argument initially:  an EmbeddingOpFactory does indeed care about `sslbls`.  If you ask it to create an operation with a given set of arguments on a set of state space labels (`sslbls`) then it will do so by creating an operation to embed from the arguments and then embedding that operation on the given `sslbls`.  My thinking was that the operation being embedded would/should never care about the state space labels it's being embedded onto -- it would be an operation for something like a continuously parameterized gate (or just a static gate) that would be the same regardless of where (e.g. on which qubits) it was placed -- it would just depend on the `args` passed in.

One could argue "well, we might as well pass it the `sslbls` since it can just ignore them if it doesn't need them", but 1) I don't think this is typically done in factories - a factory object may complain if it's handed a non-None `sslbls` and it isn't expecting any (see assertion in `UnitaryOpFactory.create_object`) and 2) it woudl be an error to give an `EmbeddedOpFactory` as the factory to embed within an `EmbeddingOpFactory` (in which case the code would attempt to embed the operation twice).  We could change the behavior of 1) and 2) isn't a big deal, so I could see proceeding with this change if we addressed the first issue.

Is the use case you're thinking of one where there's a factory that essentially creates different fixed-number-of-qubit (e.g. 1Q) gates for different sslbls?  For example, a factory that creates a simple (non-embedded) 1Q gate with over-rotation when given sslbls=(0,) and a different 1Q gate with under-rotation when sslbls=(1,)?

Overall, my immediate concern is that accepting this change will break the ability to use an `EmbeddingOpFactory` to embed a `UnitaryOpFactory` (i.e., a continuously parameterized gate) onto arbitrary qubits, and this seems like a use case we want to allow.

Reply:
I've reverted the change for now. Yes, my use case is more or less what you described - I want sslbl-specific noise on my ops. This is for the 100 qubit CHP simulations, so there are a lot of ops and we may not actually use all the individual ops in each simulation, so I was trying to build them on the fly. The EmbeddedOpFactory is completely sslbl-agnostic, so I thought maybe the EmbeddingOpFactory could be repurposed for this. Maybe we can chat about this soon and follow up with an appropriate change in a different PR then.

Reply:
Sounds good.